spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -535,12 +537,22 @@ public Scan buildChangelogScan() {
           SparkReadOptions.END_TIMESTAMP);
     }
 
-    if (startTimestamp != null) {
-      startSnapshotId = getStartSnapshotId(startTimestamp);
-    }
+    boolean emptyScan = false;
+    if (startTimestamp != null || endTimestamp != null) {
+      Pair<Long, Long> snapshots = getSnapshotsFromTimeRange(startTimestamp, endTimestamp);
```

Comment:
It seems not worth to have a separated method.  Would suggest to revert this part of change.

Reply:
reverted